tests/adapter/dbt/tests/adapter/test_incremental_unique_id.py

```diff
@@ -0,0 +1,597 @@
+        """should pass back error state when trying build an incremental
+        model whose unique key or keylist includes a column missing
+        from the incremental model"""
+        # seed_count = len(run_dbt(["seed", "--select", "seed", "--full-refresh"]))  # noqa:F841
+        # unique keys are not applied on first run, so two are needed
+        run_dbt(
+            ["run", "--select", incremental_model_name, "--full-refresh"],
+            expect_pass=True,
+        )
+        run_result = run_dbt(
+            ["run", "--select", incremental_model_name], expect_pass=False
+        ).results[0]
+
+        return run_result.status, run_result.message
+
+
+class TestIncrementalWithoutUniqueKey(IncrementalUniqueKeyBase):
+    def test__no_unique_keys(self, project):
+        """with no unique keys, seed and model should match"""
+        seed = "seed"
+        seed_rows = 8
+        incremental_model = "no_unique_key"
+        update_sql_file = "add_new_rows"
+
+        expected_fields = self.stub_expected_fields(relation=seed, seed_rows=seed_rows)
+        test_case_fields = ResultHolder(
+            *self.setup_test(seed, incremental_model, update_sql_file, project),
+            opt_model_count=None,
+            relation=incremental_model,
+        )
+
+        self.check_scenario_correctness(expected_fields, test_case_fields, project)
+
+
+class TestIncrementalStrUniqueKey(IncrementalUniqueKeyBase):
+    def test__empty_str_unique_key(self, project):
+        """with empty string for unique key, seed and model should match"""
+        seed = "seed"
+        seed_rows = 8
+        incremental_model = "empty_str_unique_key"
+        update_sql_file = "add_new_rows"
+
+        expected_fields = self.stub_expected_fields(relation=seed, seed_rows=seed_rows)
+        test_case_fields = ResultHolder(
+            *self.setup_test(seed, incremental_model, update_sql_file, project),
+            opt_model_count=None,
+            relation=incremental_model,
+        )
+
+        self.check_scenario_correctness(expected_fields, test_case_fields, project)
+
+    def test__one_unique_key(self, project):
+        """with one unique key, model will overwrite existing row"""
+        seed = "seed"
+        seed_rows = 7
+        incremental_model = "str_unique_key"
+        update_sql_file = "duplicate_insert"
+        expected_model = "one_str__overwrite"
+
+        expected_fields = self.stub_expected_fields(
+            relation=expected_model, seed_rows=seed_rows, opt_model_count=1
+        )
+        test_case_fields = ResultHolder(
+            *self.setup_test(seed, incremental_model, update_sql_file, project),
+            opt_model_count=self.update_incremental_model(expected_model),
+            relation=incremental_model,
+        )
+
+        self.check_scenario_correctness(expected_fields, test_case_fields, project)
+
+    def test__bad_unique_key(self):
```

Comment:
You don't have a project fixture on this test.

Reply:
oh thank you wasn't sure if it would still need to take in the project as it doesn't seem to call on it since it just passes the fail case.

Reply:
If you don't have a project fixture, you don't have a project to do dbt_run against.

Reply:
ah, seeing that its a unused arg threw me off but that makes sense, thanks for the context.